spec/rubocop/cop/rspec/receive_messages_spec.rb

```diff
@@ -0,0 +1,219 @@
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      before do
+        allow(user).to receive_messages(foo: baz, bar: qux)
+      end
+    RUBY
+  end
+
+  it 'registers an offense when multiple messeages stubbed on the ' \
+     'same object and another methods called between stubs' do
+    expect_offense(<<~RUBY)
+      before do
+        allow(Service).to receive(:foo).and_return(1)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `receive_messages` instead of multiple stubs on lines [5].
+        calling_some_method
+        calling_another_method
+        allow(Service).to receive(:bar).and_return(2)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `receive_messages` instead of multiple stubs on lines [2].
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      before do
+        allow(Service).to receive_messages(foo: 1, bar: 2)
+        calling_some_method
+        calling_another_method
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when multiple messeages stubbed on the ' \
+     'same object and using multiple `.and_return` arguments' do
+    expect_no_offenses(<<~RUBY)
+      before do
+        allow(Service).to receive(:foo).and_return(1, 2)
+        allow(Service).to receive(:bar).and_return(3)
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when multiple messeages stubbed on the ' \
+     'same object and using `.and_call_original`' do
+    expect_no_offenses(<<~RUBY)
+      before do
+        allow(Service).to receive(:foo).and_call_original
+        allow(Service).to receive(:bar).and_return(qux)
+        allow(Service).to receive(:baz).and_call_original
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when multiple messeages stubbed on the ' \
+     'different object`' do
+    expect_no_offenses(<<~RUBY)
+      before do
+        allow(Foo).to receive(:foo).and_return(baz)
+        allow(Bar).to receive(:bar).and_return(bar)
+        allow(Baz).to receive(:baz).and_return(foo)
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when multiple messeages stubbed on the ' \
+     'same object and same message`' do
+    expect_no_offenses(<<~RUBY)
+      before do
+        allow(Foo).to receive(:foo).and_return(baz)
+        allow(Foo).to receive(:foo).and_return(bar)
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when multiple messeages stubbed on the ' \
+     'same object with block`' do
+    expect_no_offenses(<<~RUBY)
+      before do
+        allow(Service).to receive(:foo) { baz }
+        allow(Service).to receive(:bar) { bar }
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when multiple messeages stubbed on the ' \
+     'same object and `.with` method`' do
+    expect_no_offenses(<<~RUBY)
+      before do
+        allow(Service).to receive(:foo).with(1).and_return(baz)
+        allow(Service).to receive(:foo).with(2).and_return(bar)
```

Comment:
nitpick: 
```suggestion
        allow(Service).to receive(:bar).with(2).and_return(bar)
```
Otherwise it falls under different category (stubbing on the same method)